### PR TITLE
[Behat] EmailChecker upgrades

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Related tickets | fixes #5932 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Component/Core/Test/Services/EmailChecker.php [hunk 1]
```diff
@@ -39,25 +39,31 @@ public function __construct($spoolDirectory)
...
     public function hasRecipient($recipient)
     {
+        $this->assertRecipientIsValid($recipient);
+
         $messages = $this->getMessages($this->spoolDirectory);
         foreach ($messages as $message) {
-            if (array_key_exists($recipient, $message->getTo())) {
-                return true;
+            if (!$this->isMessageTo($message, $recipient)) {
+                return false;
```
Comment (pamil) [inline src/Sylius/Component/Core/Test/Services/EmailChecker.php]: Should it really check if all messages has the given recipient?

File: src/Sylius/Behat/Context/Ui/EmailContext.php [hunk 2]
```diff
@@ -48,22 +48,25 @@ public function __construct(SharedStorageInterface $sharedStorage, EmailCheckerI
...
 
     /**
      * @Then :count email(s) should be sent to :recipient
      */
     public function numberOfEmailsShouldBeSentTo($count, $recipient)
     {
-        $this->assertEmailHasRecipient($recipient);
+        $actualMessagesCount = $this->emailChecker->countMessagesTo($recipient);
 
         Assert::eq(
```
Comment (tuka217) [inline src/Sylius/Behat/Context/Ui/EmailContext.php]: Assert::same ?

Comment (NeverResponse) [inline src/Sylius/Behat/Context/Ui/EmailContext.php]: Won't work as the `$count` we get is a string, while `$actualMessagesCount` is an int.
Ofc `$count` could be transformed to an int, but personally i think that'd be a tad of an overkill.

**Comment by michalmarcinkowski** [comment]
Thanks Janek! 👍

**Pull request merged**